components/cache/cache_pools.rst

```diff
@@ -0,0 +1,250 @@
+contents are cached in the memory of your server, which is much faster than the
+filesystem. It requires to have installed and enabled the PHP APC extension::
+
+    use Symfony\Component\Cache\Adapter\ApcuAdapter;
+
+    $cache = new ApcuAdapter($namespace = '', $defaultLifetime = 0);
+
+``namespace``
+    **type**: string, **default value**: ``''`` (an empty string)
+    The string prefixed to the keys of the items stored in this cache.
+
+``defaultLifetime``
+    **type**: integer, **default value**: ``0``
+    The default lifetime, in seconds, applied to cache items that don't define
+    their own lifetime. The default value (``0``) means an "infinite" lifetime,
+    which in this adapter ends when the web server is restarted or the APC memory
+    is deleted in any other way.
+
+Redis Cache Adapter
+~~~~~~~~~~~~~~~~~~~
+
+This adapter, similarly to APCu adapter, can increase the application performance
+very significantly, because contents are cached in the memory of your server. It
+requires to have installed Redis and have created a connection that implements
+``\Redis`` class::
+
+    use Symfony\Component\Cache\Adapter\RedisAdapter;
+
+    $cache = new RedisAdapter(\Redis $redisConnection, $namespace = '', $defaultLifetime = 0);
+
+``redisConnection``
+    **type**: ``\Redis``, **default value**: (none, this argument is mandatory)
+    The object that represents a valid connection to your Redis system.
+
+``namespace``
+    **type**: string, **default value**: ``''`` (an empty string)
+    The string prefixed to the keys of the items stored in this cache.
+
+``defaultLifetime``
+    **type**: integer, **default value**: ``0``
+    The default lifetime, in seconds, applied to cache items that don't define
+    their own lifetime. The default value (``0``) means an "infinite" lifetime,
+    which in this adapter ends when the server is restarted or the Redis memory
+    is deleted in any other way.
+
+Chain Cache Adapter
+~~~~~~~~~~~~~~~~~~~
+
+This adapter allows to combine any number of the previous adapters. Cache items
+are fetched from the first adapter which contains them. Besides, cache items are
+saved in all the given adapters, so this is a quick way of creating a cache
+replication::
+
+    use Symfony\Component\Cache\Adapter\ApcuAdapter;
+    use Symfony\Component\Cache\Adapter\ChainAdapter;
+    use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+
+    $apcCache = new ApcuAdapter();
+    $fileCache = new FilesystemAdapter();
+
+    $cache = new ChainAdapter(array($apcCache, $fileCache));
+
+The second optional argument of ``ChainAdapter`` is the ``maxLifetime`` (default
+``0``) which is the maximum lifetime of items propagated from lower adapters to
+upper ones.
+
+.. TODO: I don't understand the previous phrase, which is copied from the ChainAdapter code.
+
+Proxy Cache Adapter
+~~~~~~~~~~~~~~~~~~~
+
+.. TODO: what is this adapter useful for?
```

Comment:
Two use cases: PSR-6 interop of cache items implementations. Without it, you can't fetch a cache item from a non-symfony pool and save it in a Symfony one. The second use case if getting stats about hits/misses in the current script context (see getHits/getMisses functions).